akka-stream/src/main/scala/akka/stream/impl/io/IOSources.scala

```diff
@@ -73,3 +75,66 @@ private[akka] final class InputStreamSource(createInputStream: () ⇒ InputStrea
   override def withAttributes(attr: Attributes): Module =
     new InputStreamSource(createInputStream, chunkSize, attr, amendShape(attr))
 }
+
+/**
+ * INTERNAL API
+ * Source backed by OutputStream
+ */
+private[akka] final class OutputStreamSource(val timeout: FiniteDuration,
+                                             val attributes: Attributes, shape: SourceShape[ByteString])
+  extends SourceModule[ByteString, (OutputStream, Future[Long])](shape) {
+
+  override def create(context: MaterializationContext) = {
+    val mat = ActorMaterializer.downcast(context.materializer)
+
+    val bytesReadPromise = Promise[Long]()
+    val settings = mat.effectiveSettings(context.effectiveAttributes)
+
+    val maxInputBufferSize = settings.maxInputBufferSize
+
+    val props = OutputStreamPublisher.props(bytesReadPromise, maxInputBufferSize).withDeploy(Deploy.local)
```

Comment:
I was tempted to propose the IO dispatcher to be used for these actors... Technically it's not "file io" it could be other stuff, but it's blocking... I'd want to put anything blocking far away from the default dispatcher.

Opinions @viktorklang @jrudolph ?